src/language-js/embed.js

```diff
@@ -284,31 +285,32 @@ function replacePlaceholders(quasisDoc, expressionDocs) {
         .concat([at + placeholder])
         .concat(rest);
     }
-    const atPlaceholderIndex = parts.findIndex(
-      (part) =>
-        typeof part === "string" && part.startsWith("@prettier-placeholder")
-    );
-    if (atPlaceholderIndex > -1) {
-      const placeholder = parts[atPlaceholderIndex];
-      const rest = parts.slice(atPlaceholderIndex + 1);
-      const placeholderMatch = placeholder.match(
-        /@prettier-placeholder-(.+)-id([\S\s]*)/
-      );
-      const placeholderID = placeholderMatch[1];
-      // When the expression has a suffix appended, like:
-      // animation: linear ${time}s ease-out;
-      const suffix = placeholderMatch[2];
-      const expression = expressions[placeholderID];
 
-      replaceCounter++;
-      parts = parts
-        .slice(0, atPlaceholderIndex)
-        .concat(["${", expression, "}" + suffix])
-        .concat(rest);
-    }
-    return { ...doc, parts };
-  });
+    const replacedParts = [];
+    parts.forEach((part) => {
+      if (typeof part !== "string" || !part.includes("@prettier-placeholder")) {
+        replacedParts.push(part);
+        return;
+      }
+
+      // When we have multiple placeholders in one line, like:
+      // ${Child}${Child2}:not(:first-child)
+      const components = part.split(/@prettier-placeholder-(\d+)-id/);
+      components.forEach((component, idx) => {
+        // The placeholder is always at odd indices
+        if (idx % 2 === 0 || !component) {
```

Comment:
Why is `|| !component` needed?

Reply:
While you're at it, please remove this unneeded cloning: `const expressions = expressionDocs.slice();`. Seems to be a leftover from something.

Reply:
`|| !component` is a sanity check. I will remove it..

Okay, will do.

Reply:
Done.